File: system-test/dotnet/SystemTests/Legacy/Mod07/SmokeTests/System/ShopSmokeTest.cs
using Common;
using SystemTests.Legacy.Mod07.Base;
using Dsl.Core.Shop;
using Optivem.Testing;
using Shouldly;
using Xunit;

namespace SystemTests.Legacy.Mod07.SmokeTests.System;

public class ShopSmokeTest : BaseSystemDslTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldBeAbleToGoToShop(Channel channel)
    {
        (await (await _app.Shop(channel)).GoToShop()
            .Execute())
            .ShouldSucceed();
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/legacy/mod07/smoke/system/ShopSmokeTest.java
package com.optivem.shop.systemtest.legacy.mod07.smoke.system;

import com.optivem.shop.systemtest.legacy.mod07.base.BaseUseCaseDslTest;
import com.optivem.shop.dsl.channel.ChannelType;
import com.optivem.testing.Channel;
import org.junit.jupiter.api.TestTemplate;

class ShopSmokeTest extends BaseUseCaseDslTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldBeAbleToGoToShop() {
        app.shop().goToShop()
                .execute()
                .shouldSucceed();
    }
}

